tests/unit/lms/services/http_test.py

```diff
@@ -1,134 +1,86 @@
-        )
-
-    def test_it_sends_request_headers(self, svc, url):
-        svc.request("GET", url, headers={"HEADER_KEY": "HEADER_VALUE"})
-
-        assert httpretty.last_request().headers["HEADER_KEY"] == "HEADER_VALUE"
-
-    def test_it_sends_request_auth(self, svc, url):
-        svc.request("GET", url, auth=("user", "pass"))
-
-        assert httpretty.last_request().headers["Authorization"] == "Basic dXNlcjpwYXNz"
-
-    def test_it_uses_custom_timeouts(self, session):
-        svc = HTTPService(session)
-
-        svc.request("GET", "https://example.com", timeout=3)
-
-        assert session.request.call_args[1]["timeout"] == 3
-
-    def test_it_passes_arbitrary_kwargs_to_requests(self, svc):
-        session = Mock()
-        svc = HTTPService(session)
-
-        svc.request("GET", "https://example.com", foo="bar")
-
-        assert session.request.call_args[1]["foo"] == "bar"
-
-    @pytest.mark.parametrize(
-        "exception",
-        [
-            requests.ConnectionError(),
-            requests.HTTPError(),
-            requests.ReadTimeout(),
-            requests.TooManyRedirects(),
-        ],
-    )
-    def test_it_raises_if_sending_the_request_fails(self, exception, session):
-        session.request.side_effect = exception
-        svc = HTTPService(session)
-
         with pytest.raises(ExternalRequestError) as exc_info:
             svc.request("GET", "https://example.com")
 
-        assert exc_info.value.response is None
-        assert exc_info.value.__cause__ == exception
+        assert exc_info.value.request == sentinel.err_request
+        assert exc_info.value.response == None
 
-    @pytest.mark.parametrize("status", [400, 401, 403, 404, 500])
-    def test_it_raises_if_the_response_is_an_error(self, svc, url, status):
-        httpretty.register_uri("GET", url, status=status)
+    def test_it_raises_if_the_response_is_an_error(self, svc):
+        response = svc.session.request.return_value
+        response.raise_for_status.side_effect = RequestException(
+            request=sentinel.err_request, response=sentinel.err_response
+        )
 
         with pytest.raises(ExternalRequestError) as exc_info:
-            svc.request("GET", url)
+            svc.request("GET", "https://example.com")
 
-        assert isinstance(exc_info.value.__cause__, requests.HTTPError)
-        assert exc_info.value.response == Any.instance_of(requests.Response).with_attrs(
-            {"status_code": status}
-        )
+        assert exc_info.value.request == sentinel.err_request
+        assert exc_info.value.response == response
 
-    @pytest.fixture
-    def url(self):
-        """Return the URL that we'll be sending test requests to."""
-        return "https://example.com/example"
-
-    @pytest.fixture(autouse=True)
-    def test_response(self, url):
-        httpretty.register_uri(
-            "GET", url, body='{"test_response_key": "test_response_value"}', priority=-1
-        )
```

Comment:
There's an argument to say these aren't great tests, because if requests changes in such a way as this breaks, perhaps we wouldn't notice.

Some observations:

 * The mock will keep us mostly straight on the acceptable args
 * We don't really do much here, so there's not much interface to get wrong
 * If requests is broken... we'll notice, and almost certainly have bigger fish to fry